src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -615,6 +615,11 @@ def commandline(self, id_):
                     port = rsync['remoteport']
                     extra_args = ''
 
+                remote_username, remote_host = remote.rsplit('@', 1)
+                if ':' in remote_host:
```

Comment:
Do we know that the `remote_host` doesn't already have enclosing brackets?

Reply:
Yes, it's a valid hostname